packages/flame/lib/src/game/flame_game.dart

```diff
@@ -118,4 +118,20 @@ class FlameGame extends Component with Game {
 
   @override
   Projector get projector => camera.combinedProjector;
+
+  /// Only use this method for tests, it makes sure that the [component]s
+  /// added to the tree if you wait for the returned future to resolve.
+  @visibleForTesting
+  Future<void> addUpdate(Component component) async {
```

Comment:
What if we add this to the tester instead? Maybe as an extension

Reply:
Sounds good, but how can it access things from there inside the lamda in which we are running the test? Or you mean like a static function?

Reply:
Added them as static functions now, but if there is a way to make the function accessible within the lambda without being static, please do tell. :smile:

Reply:
This is a method on the flame game class right? So it can be an extension method, but declared on flame test dependency. That way we don't need this comment and the visible for testing annotation

Reply:
An extension on the game class? Then we need to add that extension every time we make a new game class within a test file?

Reply:
Haha sorry, I'm tired, that is genius of course that works. I'll do that. :)